tests/integration/learning/test_fleet_state.py

```diff
@@ -87,41 +98,51 @@ def test_state_is_recorded_after_learning(federated_ursulas, lonely_ursula_maker
     """
     _lonely_ursula_maker = partial(lonely_ursula_maker, quantity=1)
     lonely_learner = _lonely_ursula_maker().pop()
+    states = lonely_learner.known_nodes._archived_states
 
     # This Ursula doesn't know about any nodes.
     assert len(lonely_learner.known_nodes) == 0
 
     some_ursula_in_the_fleet = list(federated_ursulas)[0]
     lonely_learner.remember_node(some_ursula_in_the_fleet)
-    assert len(lonely_learner.known_nodes.states) == 1  # Saved a fleet state when we remembered this node.
+    assert len(states) == 2  # Saved a fleet state when we remembered this node.
+
+    # The first fleet state is just us and the one about whom we learned, which is part of the fleet.
+    assert states[-1].population == 2
 
     # The rest of the fucking owl.
     lonely_learner.learn_from_teacher_node()
 
-    states = list(lonely_learner.known_nodes.states.values())
-    assert len(states) == 2
+    # There are two new states: one created after seednodes are loaded, to select a teacher,
+    # and the second after we get the rest of the nodes from the seednodes.
+    assert len(states) == 4
 
-    assert len(states[0].nodes) == 2  # The first fleet state is just us and the one about whom we learned, which is part of the fleet.
-    assert len(states[1].nodes) == len(federated_ursulas) + 1  # When we ran learn_from_teacher_node, we also loaded the rest of the fleet.
+    # When we ran learn_from_teacher_node, we also loaded the rest of the fleet.
+    assert states[-1].population == len(federated_ursulas) + 1
 
 
 def test_teacher_records_new_fleet_state_upon_hearing_about_new_node(federated_ursulas, lonely_ursula_maker):
     _lonely_ursula_maker = partial(lonely_ursula_maker, quantity=1)
     lonely_learner = _lonely_ursula_maker().pop()
 
     some_ursula_in_the_fleet = list(federated_ursulas)[0]
+
     lonely_learner.remember_node(some_ursula_in_the_fleet)
 
+    states = some_ursula_in_the_fleet.known_nodes._archived_states
 
-    teacher_states_before = list(some_ursula_in_the_fleet.known_nodes.states.values())
+    states_before = len(states)
     lonely_learner.learn_from_teacher_node()
-    teacher_states_after = list(some_ursula_in_the_fleet.known_nodes.states.values())
+    states_after = len(states)
+
+    # FIXME: some kind of a timeout is required here to wait for the learning to end
```

Comment:
I'm sensing that `FIXME` is overused in the code base.  Perhaps change this to a `TODO` and open an Issue to track.

Reply:
Filed #2568